unblob/dependencies.py

```diff
@@ -0,0 +1,57 @@
+from shutil import which
+from typing import Dict, List, Tuple
+
+from click import Context, Option, echo
+
+from .handlers import ALL_HANDLERS_BY_PRIORITY
+
+INSTALLED = "\u2713"
+NOT_INSTALLED = "\u2717"
+
+
+def get_dependencies() -> List[str]:  # pragma: no cover
+    return sorted(
+        set(
+            [
+                h._get_extract_command()
+                for handlers in ALL_HANDLERS_BY_PRIORITY
+                for h in handlers
+            ]
+        )
+    )
+
+
+def _is_installed(command: str) -> bool:
+    return bool(which(command))
+
+
+def _get_dependency_map() -> Dict[str, bool]:
+    return {cmd: _is_installed(cmd) for cmd in get_dependencies()}
+
+
+def _pretty_format_dependencies() -> Tuple[int, str]:
+    return_code = 0
+    dependency_map = _get_dependency_map()
+    longest_legth = len(max(dependency_map.keys(), key=len))
+    text = "The following executables found installed, which are needed by unblob:"
```

Comment:
Python text concatenation 1x1: put the lines in a list and join them with new line. It's a more idiomatic way. I think I learned it from here: https://waymoot.org/home/python_string/ but there are many newer articles why it's better:
https://dock2learn.com/tech/how-to-efficiently-concatenate-strings-in-python/
https://blog.ganssle.io/articles/2019/11/string-concat.html
https://betterprogramming.pub/python-and-string-concatenation-31772a10fed
https://grski.pl/fstrings-performance.html

In this specific case, the speed wouldn't matter at all, but I find still easier to read `"\n".join(lines)` than figuring out in my head where and when to put new line characters.